spec/concerns/queue_it/queable_by_name_spec.rb

```diff
@@ -0,0 +1,107 @@
+describe 'Concerns::QueableByName' do
+  let(:task) { create(:job) }
+
+  describe '#push_to_queue' do
+    context 'add node to queue' do
+      let(:nodable) { create(:user) }
+
+      it 'expect queue to not be empty' do
+        expect(task.queues.count).to eq(0)
+        task.push_to_queue("my_users", nodable)
+        expect(task.queues.count).to eq(1)
+      end
+    end
+  end
+
+  describe '#pop_from' do
+    it "should return the first node in the queue sorted by field" do
+      task.push_to_queue("cops", create(:user, name: "John"))
+      task.push_to_queue("cops", create(:user, name: "Bob"))
+
+      result = task.pop_from(
+        queue_name: "cops",
+        sort_exp: :name,
+        sort_order: :desc
+      )
+      expect(result.name).to eq("Bob")
+    end
+
+    it "should return the first node in the queue sorted by expression" do
+      task.push_to_queue("cops", create(:user, name: "John"))
+      task.push_to_queue("cops", create(:user, name: "Ana"))
+
+      result = task.pop_from(
+        queue_name: "cops",
+        sort_exp: ->(u) { u.name.length },
+        sort_order: :desc
+      )
+      expect(result.name).to eq("Ana")
+    end
+
+    it "should return the first node in the queue sorted by expression that uses data defined outside the node" do
+      task.push_to_queue("cops", create(:user, name: "Eric"))
+      task.push_to_queue("cops", create(:user, name: "Xia"))
+      task.push_to_queue("cops", create(:user, name: "John"))
+
+      dobs = {
+        "Xia" => Date.new(1992, 1, 1),
+        "Eric" => Date.new(1990, 1, 1),
+        "John" => Date.new(1991, 1, 1)
+      }
+
+      result = task.pop_from(
+        queue_name: "cops",
+        sort_exp: ->(u) { dobs[u.name] },
+        sort_order: :desc
+      )
+      expect(result.name).to eq("Eric")
```

Comment:
trying to understand this one. if the sort is desc, then is it youngest to oldest? If so, wouldn't the first be "Xia"?

Reply:
nevermind, i just remembered that you had mentioned wanting to pop from the tail instead of the head